nfsv4/client.c

```diff
@@ -35,32 +35,40 @@ char *nfs4_error_string(nfs4 n)
 // the error is recorded in the nfs4 
 int nfs4_pread(nfs4_file f, void *dest, bytes offset, bytes len)
 {
-    u64 total = 0;
+    u64 total_requested = 0;
+
+    u64 total_sent = 0;
+    u32 eof = 0;
 
     while (1) {
         rpc r = file_rpc(f);
-        u64 transferred = push_read(r, offset,  dest + total, len - total, &f->latest_sid);
+        
+        struct nfs4_read_data *read_data = malloc(
```

Comment:
Replace this with a variable on the stack that contains static pointers to `eof`, `total_sent`, `dest`, `total_requested`